crd/src/util.rs

```diff
@@ -308,8 +318,8 @@ fn get_opa_connection_string_from_pods(
     // changes to the infrastructure
     server_and_port_list.sort_by(|(host1, _), (host2, _)| host1.cmp(host2));
 
-    // TODO: randomly search for node? -> for now we just take the last one
-    let server = server_and_port_list.pop();
+    let index = rand::random::<usize>() % server_and_port_list.len();
+    let server = server_and_port_list.get(index);
```

Comment:
There's a method for this: https://stackoverflow.com/questions/34215280/how-can-i-randomly-select-one-element-from-a-vector-or-array/34215930